### PR TITLE
feat(docker): Support more Dockerfile filename variants

### PR DESCRIPTION
**Pull request by elliotwaite**
This adds Dockerfile support for `*.dockerfile`, `Dockerfile.dev`, and `Dockerfile.prod`. [This GitHub comment](https://github.com/microsoft/vscode-docker/issues/1907#issuecomment-618477074) mentions that these are the top three alternative Dockerfile filename variants based on VS Code's telemetry data.

I wasn't sure where to add the "dockerfile" entry in the "fileExtensions" section, so I just added it to the end. Let me know if I should add it somewhere else. Also, my editor auto-fixed the formatting of the items above the new "dockerfile" entry (removing the extra spaces to the left of the colons), but I can undo those fixes if you would prefer that I don't include those changes in this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "email": "me@chadalen.com",
     "url": "https://cadamsdev.com"
   },
-  "version": "1.1.0",
+  "version": "1.1.1",
```
Comment (cadamsdev) [inline package.json:11]: You don't have to revert this change but just FYI you don't need to update this. I have semantic-release setup which should handle creating the new version automatically as long as there is a valid commit message. (see [here](https://github.com/cadamsdev/vscode-jetbrains-icon-theme/blob/main/docs/CONTRIBUTING.md#commit-messages))

**Comment by cadamsdev** [comment]
Looks good to me. Thanks for contributing.

File: package.json (hunk 1 above)
Comment (elliotwaite) [reply package.json:11]: Ah, okay, got it.
